lib/index.js

```diff
@@ -2,42 +2,66 @@ import duplex from 'duplexer'
 import concat from 'concat-stream'
 import { PassThrough } from 'readable-stream'
 import Canvas from 'canvas'
-import brick from './brick'
 
-const getAverageColor = data => data
+const brick = __dirname + '/../images/brick.png'
+const defaultOpts = { format: 'jpeg', mode: 'good', brick }
+const pixelIntervalPreset = { fast: 8, good: 5, best: 3, nearest: 2, bilinear: 1 }
+
+// Make a function to get the average color from given image data array
+// with given sample rate.
+const makeGetAverageColor = pixelInterval => data => data
+  .filter((_, index) => index % (pixelInterval * 4) < 4)
   .reduce((rgb, channel, index) => (rgb[index % 4] += channel, rgb), [0, 0, 0])
   .map(channel => Math.floor(channel / (data.length / 4)))
 
-const renderBrick = (ctx, x, y) => {
-  const color = getAverageColor(ctx.getImageData(x, y, brick.width, brick.height).data)
+// Make a function to render a brick at a point using given brick, average
+// color function and context.
+const makeRenderBrick = (brick, getAverageColor, ctx) => (x, y) => {
```

Comment:
So we can do `const renderBrick = makeRenderBrick(brick, getAverageColor, ctx)` and use the `renderBrick` function taking `x` and `y` only.